e2e/nomostest/helm/helm.go

```diff
@@ -18,174 +18,284 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
-	"strconv"
 	"strings"
-	"time"
 
+	"github.com/ettle/strcase"
+	"github.com/google/uuid"
 	"kpt.dev/configsync/e2e"
 	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/testlogger"
 	"kpt.dev/configsync/e2e/nomostest/testshell"
 	"sigs.k8s.io/kustomize/kyaml/copyutil"
+	"sigs.k8s.io/yaml"
 )
 
-// PrivateARHelmRegistry is the registry URL to the private AR used for testing.
-// Cannot be assigned as a global variable due to GCPProject pointer, which gets
-// evaluated after initialization.
-func PrivateARHelmRegistry() string {
-	return fmt.Sprintf("oci://us-docker.pkg.dev/%s/config-sync-test-ar-helm", *e2e.GCPProject)
-}
-
-// PrivateARHelmHost is the host name of the private AR used for testing
-var PrivateARHelmHost = "https://us-docker.pkg.dev"
-
 // RemoteHelmChart represents a remote OCI-based helm chart
 type RemoteHelmChart struct {
 	// Shell is a helper utility to execute shell commands in a test.
 	Shell *testshell.TestShell
 
-	// Host is the host URL, e.g. https://us-docker.pkg.dev
-	Host string
+	// Logger to write logs to
+	Logger *testlogger.TestLogger
+
+	// Project in which to store the chart image
+	Project string
 
-	// Registry is the registry URL, e.g. oci://us-docker.pkg.dev/oss-prow-build-kpt-config-sync/config-sync-test-ar-helm
-	Registry string
+	// Location to store the chart image
+	Location string
+
+	// RepositoryName in which to store the chart imageS
+	RepositoryName string
 
 	// ChartName is the name of the helm chart
 	ChartName string
 
 	// ChartVersion is the version of the helm chart
 	ChartVersion string
 
-	// Dir is a local directory from which RemoteHelmChart will read, package, and push the chart from
-	Dir string
+	// LocalChartPath is a local directory from which RemoteHelmChart will read, package, and push the chart from
+	LocalChartPath string
 }
 
-// NewRemoteHelmChart creates a RemoteHelmChart
-func NewRemoteHelmChart(shell *testshell.TestShell, host, registry, dir, chartName, version string) *RemoteHelmChart {
-	return &RemoteHelmChart{
-		Shell:        shell,
-		Host:         host,
-		Registry:     registry,
-		Dir:          dir,
-		ChartName:    chartName,
-		ChartVersion: version,
+// CreateRepository uses gcloud to create the repository, if it doesn't exist.
+func (r *RemoteHelmChart) CreateRepository() error {
+	out, err := r.Shell.ExecWithDebug("gcloud", "artifacts", "repositories",
+		"describe", r.RepositoryName,
+		"--location", r.Location)
+	if err != nil {
+		if !strings.Contains(string(out), "NOT_FOUND") {
+			return fmt.Errorf("failed to describe image repository: %w", err)
+		}
+		// repository does not exist, continue with creation
+	} else {
+		// repository already exists, skip creation
+		return nil
+	}
+
+	r.Logger.Info("Creating image repository")
+	_, err = r.Shell.ExecWithDebug("gcloud", "artifacts", "repositories",
+		"create", r.RepositoryName,
```

Comment:
This requires the CI service account (`e2e-test-runner`) to have the `Artifact Registry Administrator` role to create repositories.
The SA is not managed by latchkey yet, so we need to update it in both the main project and release project.

Reply:
I have updated the IAM in stolos-dev and oss-prow-build-kpt-config-sync to replace Writer with Admin.

Reply:
Those two are the wrong projects, I can take a look

Reply:
We need to update `kpt-config-sync-ci-main` and `kpt-config-sync-ci-release`.
`stolos-dev` only has monorepo CI jobs, which doesn't support OCI and Helm.
`oss-prow-build-kpt-config-sync` runs the prow jobs, but the job creates the registry in the main and release projects.